### PR TITLE
Review and reorganize CMakeLists.txt file

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

This PR is improving the readability of `CMakeLists.txt` basically by:
- Grouping similar entries
- Organizing the output messages
- Reviewing comments

#### Rationale:

Improve readability of `CMakeLists.txt` file to make easier to work on it.

#### Review Hints:

Besides the readability of the file and some changes in the output, the test is basically building the products and confirming everything that it still working as expected.

### CODE REVIEW

**Review by jan-cerny — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -536,13 +530,11 @@ The %{name} package also contains HTML formatted documents containing
...
+# %{release} includes release number and dist type.
+# This only has effect with cmake v3.6 or higher.
 set(CPACK_RPM_FILE_NAME "%{name}-%{version}-%{release}.rpm")
-# For older versions of cmake (e.g. v2.8) file name is defined like below
+# For older versions of cmake (e.g. v2.8) file name is defined like below.
 set(CPACK_PACKAGE_FILE_NAME "scap-security-guide-${SSG_VERSION}")
 set(CPACK_PACKAGE_CHECKSUM SHA512)
-
 set(CPACK_GENERATOR "RPM;DEB")
-include(CPack)
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply CMakeLists.txt:548]: Done. Thanks

File: CMakeLists.txt [hunk 2]
```diff
@@ -8,75 +19,87 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
...
-
-set(SSG_CONTENT_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/xml/scap/ssg/content")
-set(SSG_GUIDE_INSTALL_DIR "${CMAKE_INSTALL_DOCDIR}/guides")
-set(SSG_TABLE_INSTALL_DIR "${CMAKE_INSTALL_DOCDIR}/tables")
-set(SSG_ANSIBLE_ROLE_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/scap-security-guide/ansible")
-set(SSG_ANSIBLE_PER_RULE_PLAYBOOKS_INSTALL_DIR "${SSG_ANSIBLE_ROLE_INSTALL_DIR}/rule_playbooks")
-set(SSG_BASH_ROLE_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/scap-security-guide/bash")
-set(SSG_KICKSTART_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/scap-security-guide/kickstart")
-set(SSG_TAILORING_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/scap-security-guide/tailoring")
+cmake_dependent_option(ENABLE_PYTHON_COVERAGE "Enable Python tests with coverage support" ON "PY_PYTEST_COV" OFF)
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply CMakeLists.txt]: Done. Thanks

**Comment by jan-cerny** [comment]
@marcusburghardt Please have a look into the failing test `testing-farm:centos-stream-8-x86_64:/static-checks`. I have rerun it but it failed again. The error message hints that it might be legitimate:

```
-- Find Python Modules:
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PY_jinja2 (missing: PY_JINJA2)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPythonModule.cmake:40 (find_package_handle_standard_args)
  CMakeLists.txt:150 (find_python_module)
```

**Comment by marcusburghardt** [comment]
> @marcusburghardt Please have a look into the failing test `testing-farm:centos-stream-8-x86_64:/static-checks`. I have rerun it but it failed again. The error message hints that it might be legitimate:
> 
> ```
> -- Find Python Modules:
> CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
>   Could NOT find PY_jinja2 (missing: PY_JINJA2)
> Call Stack (most recent call first):
>   /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
>   cmake/FindPythonModule.cmake:40 (find_package_handle_standard_args)
>   CMakeLists.txt:150 (find_python_module)
> ```

I could not reproduce it locally. I need to investigate it more in a VM similar to CI tests. Moved to Draft for now.

**Comment by marcusburghardt** [comment]
I believe the issue is now fixed in the last commit. Just waiting the CI tests to confirm.
More details in the commit message.

**Pull request merged**